Fully qualify container images

eerhardt opened the pull request:
Podman doesn't default to the docker.io registry when no registry is listed. Fully qualify the container image and registry to avoid any issues with assuming a reference.

Fix #3423
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3500)

eerhardt committed:
Fully qualify container images

Podman doesn't default to the docker.io registry when no registry is listed. Fully qualify the container image and registry to avoid any issues with assuming a reference.

Fix #3423

mitchdenny commented on src/Aspire.Hosting.Kafka/KafkaBuilderExtensions.cs:
```diff
@@ -26,6 +26,7 @@ public static IResourceBuilder<KafkaServerResource> AddKafka(this IDistributedAp
         return builder.AddResource(kafka)
             .WithEndpoint(targetPort: KafkaBrokerPort, port: port, name: KafkaServerResource.PrimaryEndpointName)
             .WithImage("confluentinc/confluent-local", "7.6.0")
+            .WithImageRegistry("docker.io")
```
Capture this in a class like the other ones?

eerhardt replied:
Done.

eerhardt committed:
Address PR feedback

- Be consistent with all hosting extensions and split the ImageTags into a separate static class
- Fix tests

eerhardt committed:
Merge remote-tracking branch 'upstream/main' into FixPodman

davidfowl commented:
/azp run

eerhardt committed:
Correct Azure ContainerImageAnnotation usages.

davidfowl commented:
/backport to release/8.0
